Add article: Authenticate to third-party services (#420)
microsoft/vcpkg-docs: vcpkg/concepts/asset-caching.md

author: vicroms
ms.author: viromer
ms.topic: concept-article
ms.date: 10/03/2024
---
# Asset caching with vcpkg

[!INCLUDE [experimental](../../includes/experimental.md)]

A fundamental operation of vcpkg is downloading files from the Internet, things like: source code archives, tool executables,

> source code patches, etc. We call anything that is downloaded from the network an **asset**.

Reviewer: Are git-based registries an asset (if they are downloaded from the internet)?

Reviewer: Keywords to consider: simple downloads with known hashes, HTTP(S)

Asset caching allows you to configure alternative download locations (caches) to acquire assets. When enabled, vcpkg first

> attempst to acquire assets from any configured cache location, and falls back to the original download if no cached asset

Reviewer: attempts

is found. Optionally, vcpkg stores the asset in the cache for future use.

This feature helps mitigate these common development concerns:

> * Reducing dependency from external networks. Possibly, also reducing download times.
> * Mirroring assets in trusted locations for air-gapped environments.
> * Maintaining continuity of business if assets hosted by third-parties are changed or unavailable.

Reviewer: ```suggestion
* Reducing download times and bandwidth costs.
* Mirroring assets in trusted locations for air-gapped or offline environments.
* Maintaining continuity of business if assets hosted by third-parties are changed or unavailable.
```

* The `X_VCPKG_ASSET_SOURCES` environment variable, or
* The `--x-asset-sources` command-line option.

The expected value is a semicolon-delimited list of asset caching sources. Read the [asset caching reference
documentation](../users/assetcaching.md) to learn the configuration syntax.

> Example using Azure Blob Storage as an asset cache:

Reviewer: ```suggestion
### Example using Azure Blob Storage:
```
